dotNETStandard/HealthVault.Mobile/cs/Xamarin.Core/ViewModels/MenuViewModel.cs

```diff
@@ -125,10 +126,31 @@ public class MenuViewModel : ViewModel
             IThingClient thingClient = this.connection.CreateThingClient();
             HealthRecordInfo record = person.SelectedRecord;
             IReadOnlyCollection<Medication> items = await thingClient.GetThingsAsync<Medication>(record.Id);
+            IVocabularyClient vocabClient = this.connection.CreateVocabularyClient();
+            var ingredientChoices = new List<VocabularyItem>();
+
+            Vocabulary ingredientVocabulary = null;
+            while (ingredientVocabulary == null || ingredientVocabulary.IsTruncated)
+            {
+                string lastCodeValue = null;
+                if (ingredientVocabulary != null)
+                {
+                    lastCodeValue = ingredientVocabulary.Values.Last().Value;
```

Comment:
I don't think it should happen since IsTruncated should be false and prevent a call that would return 0 items. But I'll break from the loop if there is an empty collection.

---
In reply to: [117124010](https://github.com/Microsoft/healthvault-samples/pull/8#discussion_r117124010) [](ancestors = 117124010)